server/src/main/java/org/opensearch/extensions/ExtensionActionListener.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction.Response;
+
+/**
+ * ActionListener for ExtensionsManager
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListener implements ActionListener<Response> {
```

Comment:
This class still puzzles me: 
 - we only log here and nothing else (nothing actionable)
 - the `exceptionList` is collected but never used (except is tests)

Why do we need it?

Reply:
The intent of this class is to mirror AD functionality for ActionListener onFailure(), since the extension will no longer have access to ActionListeners from OpenSearch.  Therefore, this class does exactly what happens in the plugin onFailure() currently.

Reply:
> The intent of this class is to mirror AD functionality for ActionListener onFailure()

Apologies, what is "AD functionality"?

> since the extension will no longer have access to ActionListeners from OpenSearch.

I didn't get this part, the class implements `ActionListener` from OpenSearch: `ExtensionActionListener implements ActionListener<Response>`

>  Therefore, this class does exactly what happens in the plugin onFailure() currently.

I think we should not replicate if it does not bring any value: why we are collection "exceptionList" if no one cares? (apologies If I am missing something, please point out where the results of the `ExtensionActionListener` are being consumed or/and acted upon)

Reply:
That makes sense.  I'll remove it from the PR